lib/Dialect/Linalg/Transforms/Promotion.cpp

```diff
@@ -189,20 +189,22 @@ static void promoteSubViewOperands(LinalgOp op, SetVector<Value *> subViews,
   // extra scalars etc.
   auto operands = getAssumedNonViewOperands(op);
   opViews.append(operands.begin(), operands.end());
-  op.clone(b, op.getLoc(), opViews);
+  LinalgOp res = op.clone(b, op.getLoc(), opViews);
 
   // 3. Emit write-back for the promoted output views: copy the partial view.
   for (auto viewAndPartialLocalView : writebackViews) {
     // Note: use the old op to determine whether the operand view is an output.
     bool isOutput =
-        op.getIndexOfOutput(viewAndPartialLocalView.first).hasValue();
+        res.getIndexOfOutput(viewAndPartialLocalView.first).hasValue();
```

Comment:
This is a bug that prevent writebacks from happening.
As the comment right above says, you want to check the old op not the newly cloned one.
I'll make it into a WARNING.

Fixing internally for submitting.